src/assets/Keycloak/Keycloak.md

```diff
@@ -0,0 +1,147 @@
+Additionally, it enables users to authenticate through various identity providers and use fine-grained permissions for regulating access to Software as a Service (SaaS) applications.
+
+It facilitates the creation of a user database with customizable roles and groups, offering functionalities such as user management, registration, and password policy enforcement.
+This makes it a comprehensive Identity and Access Management solution for contemporary applications. Keycloak API enables the integration of all these features into your application
+without the need for additional coding.
+
+## Keycloak Features
+![Keycloak features](/src/assets/Keycloak/images/Keycloak/Keycloak-features.png)
+
+1. Identity Broker:
+   Identity Broker acts as a middleman, linking various service providers with different identity providers. It establishes trust with external identity providers to utilize their identities for accessing internal services offered by service providers.
+   Imagine a textile company, which has two Keycloak setups: one for customer and another for internal employees. If you want your employees to access external services without creating new accounts on customer Keycloak, you can use the customer Keycloak
+   as `Broker` and the internal Keycloak as `Provider` in this scenario.
+
+2. User Federation:
+   Keycloak provides the ability to integrate with existing LDAP or Active Directory servers. It also supports implementation of your own provider such as a relational database.
+
+3. Standard Protocols:
+   Keycloak is based on three standard protocols i.e., OpenID Connect, OAuth2.0, and SAML.
+
+4. Password Policies:
+   Keycloak contains different password policies i.e., HashAlgorithm, Hashing Iterations, Digits, Lowercase Characters, Regular Expression and so on.
+
+5. Single Sign-On:
+   Keycloak facilitates a seamless login experience where users only need to sign in once to access multiple applications, eliminating the need for repeated logins.
+   This not only streamlines user experience but also simplifies identity management for administrators.
+
+6. Social Login:
+   Keycloak has built-in support to login via Google, GitHub, and Facebook, which helps us to use social identity providers.
+
+## Keycloak Drawbacks
+Despite its extensive features, Keycloak does have certain limitations. One of these is the need for a more varied implementation approach.
+Additionally, the following are some of the drawbacks associated with Keycloak:
+1. Complex Server Deployment: The manual process involved in deploying Keycloak on a server can be complicated, potentially affecting overall productivity.
+2. Lack of Comprehensive Support Documentation: Despite its robust functionalities, Keycloak could benefit from more efficient and extensive support documentation. This would make it easier for users to find suitable solutions to their problems.
+
+## Keycloak Installation
+### By Docker image
```

Comment:
use the same styles for all titles, either all words starting with capital letters or just the first word (and names or other words that normally are written with capital letters)

Reply:
Didn't I follow the same naming rule? Sry can't get what you mean ?

Reply:
In English usually all words of a sentence are written with small letters, just the first word and names are written with capitals. Some people use capitals for all words in titles, the style you applied in line 49. In line 50 however, you only used capital for "By" (first word of the line) and "Docker" (name), but not "image"... 
Same applies to some other titles throughout the blog post. It's up to you to choose which style you prefer, but use the same title style for all your titles :)